test/CMakeLists.txt

```diff
@@ -1,3 +1,41 @@
-add_test(NAME test_check_yaml_keys
+##### get test data from EMC FTP server
+# set URL and hash
+set(URL "https://ftp.emc.ncep.noaa.gov/static_files/public/GDASApp")
+set(SHA "75c4ce84028956c1ce050e07f70abc25c540492c3e1a40f6fa56d35182322e8e")
+string(SUBSTRING ${SHA} 0 6 SHORTSHA)
+set(TAR "gdasapp-fix-${SHORTSHA}.tgz")
+# download test files
+file(DOWNLOAD
+  ${URL}/${TAR}
+  ${CMAKE_CURRENT_BINARY_DIR}/${TAR}
+  INACTIVITY_TIMEOUT 30
+  TIMEOUT 90
+  SHOW_PROGRESS
+  STATUS status
+  EXPECTED_HASH SHA256=${SHA}
+  )
+# Extract downloaded tarball.
+file(ARCHIVE_EXTRACT INPUT ${TAR})
```

Comment:
Need to change the cmake min version to something more recent. 3.16.3  is too old for the option above, 3.20.0 seems fine.